### PR TITLE
Add custom asset rights to profile form

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Show custom asset rights in a new "Custom assets" section on the Asset tab in the Profile forms.

### CODE REVIEW

File: src/Profile.php [hunk 1]
```diff
@@ -1169,6 +1169,12 @@ public static function getRightsForForm(string $interface = 'all', string $form
                     ]
                 ]
             ];
+
+            // Add rights for custom assets
+            $definitions = \Glpi\Asset\AssetDefinitionManager::getInstance()->getDefinitions();
+            foreach ($definitions as $definition) {
```
Comment (cedric-anne) [inline src/Profile.php:1175]: @orthagh Should we display inactive definitions here?

**Review by cedric-anne — CHANGES_REQUESTED** [review]
The profile matrix should not be saved directly in the `glpi_profilerights` table. They must be stored in the `profiles` field of each definition, and then the `AssetDefinition::syncProfilesRights()` method will update the profile rights depending on the asset definition state.

This buffer field was necessary as it is not possible to store right of inactive definition inside the `glpi_profilerights` table with out actually giving access to them.

File: src/Profile.php (hunk 1 above)
Comment (orthagh) [reply src/Profile.php:1175]: I'm not sure. I'm inclined on no.

**Comment by cconard96** [comment]
> The profile matrix should not be saved directly in the `glpi_profilerights` table. They must be stored in the `profiles` field of each definition, and then the `AssetDefinition::syncProfilesRights()` method will update the profile rights depending on the asset definition state.
> 
> This buffer field was necessary as it is not possible to store right of inactive definition inside the `glpi_profilerights` table with out actually giving access to them.

Is this not covered by overriding `canView` in each of the generated classes? It prevents the view of the related asset forms. They only thing that would need added I think is an active check on `front/asset/asset.php` where it already loads the related definition.

**Comment by cedric-anne** [comment]
> > The profile matrix should not be saved directly in the `glpi_profilerights` table. They must be stored in the `profiles` field of each definition, and then the `AssetDefinition::syncProfilesRights()` method will update the profile rights depending on the asset definition state.
> > This buffer field was necessary as it is not possible to store right of inactive definition inside the `glpi_profilerights` table with out actually giving access to them.
> 
> Is this not covered by overriding `canView` in each of the generated classes (asset, type, model)? It prevents the view of the related asset forms. They only thing that would need added I think is an active check on `front/asset/asset.php` where it already loads the related definition.

I was the first solution I implemented, but it was not possible to rely only on that, because `Session::checkRight()` and `Session::haveRight()` relies only on the `glpi_profilerights` table contents.

**Review by cedric-anne — CHANGES_REQUESTED** [review]
The form located in the asset definition `Profiles` tab is broken. Whe I try to submit it, it fails with an error: `Uncaught Exception TypeError: Glpi\Asset\AssetDefinition::getRightsForProfile(): Return value must be of type int, array returned in /var/www/glpi/src/Asset/AssetDefinition.php at line 719`.

I am not against changing the `profiles` field format, but we should keep a `validateProfileArray()` method that ensures that the provided data is compliant with the expected format.

**Comment by cconard96** [comment]
> The form located in the asset definition `Profiles` tab is broken. Whe I try to submit it, it fails with an error: `Uncaught Exception TypeError: Glpi\Asset\AssetDefinition::getRightsForProfile(): Return value must be of type int, array returned in /var/www/glpi/src/Asset/AssetDefinition.php at line 719`.
> 

Replay the last update. There is a dev migration the should fix the format of rights in the asset definition table.

**Comment by cedric-anne** [comment]
I made some changes, mainly:
 - moving the convertion from the rights matrix to the expected format into the front file, and put back the profiles input validation in the AssetDefinition class, in order to be able to use the same structure whether we use `AssetDefinition::update()` or `$DB->update()`;
 - delegating the profilerights updates to the `AssetDefinition` itself, in order to give the ability to the plugins to always catch profiles updates in their hooks.

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**